util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/interpolation/Interpolation.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.util.geometry.interpolation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * Base class of interpolation
+ * 
+ * @author Arun
+ * @param <T> the type of interpolated Geometry
+ */
+public abstract class Interpolation<T extends Geometry> {
+
+	/**
+	 * Grid step size factor
+	 */
+	protected final int GRID_FACTOR = 2;
+	/**
+	 * Next coordinate distance factor.
+	 */
+	protected final double NEXT_COORDINATE_DISTANCE_FACTOR = 4;
```

Comment:
this is the moving distance factor from one arc coordinate to find neighbour arc coordinate. Algorithm start with the distance (``NEXT_COORDINATE_DISTANCE_FACTOR * e ``) to search for the  next most deserved neighbour.